starter-code/js/index.js

```diff
@@ -1,11 +1,50 @@
 var $cart = document.querySelector('#cart tbody');
 var $calc = document.getElementById('calc');
+let finalS = 0;
+let $delB = document.querySelectorAll(".btn-delete")
+
 
 function updateSubtotal($product) {
   // Iteration 1.1
+  const price = $product.querySelector(".pu span").innerHTML;
+  const quantity = $product.querySelector(".qty label input").value;
+  let result = price * quantity;
+  $product.querySelector(".subtot span").innerHTML = result;
+  finalS += result;
+  return result;
 }
 
 function calculateAll() {
   // Iteration 1.2
+  let products = document.querySelectorAll(".product");
+  products.forEach(p => updateSubtotal(p))
+  document.getElementById("toto").innerHTML = finalS;
+  finalS= 0;
 }
-$calc.onclick = calculateAll;
\ No newline at end of file
+
+$calc.onclick = calculateAll;
+
+$delB.forEach(dbtn => {
+  dbtn.onclick = (e) => {
+  e.target.parentElement.parentElement.remove()
+  calculateAll();
+}})
+
+
+//create onclick event listener on create button
+document.getElementById('create').onclick = create;
+
+function create () {
+ 
+  // add this tr to the bottom of $cart
+  let row = $cart.insertRow(-1)
+
+  // create new tr with 5 td elements (name, pu, qty, subt, rm)
+  // set name = name and pu = pu
+  let newName = document.getElementById("new-name").value;
+  let newPu = document.getElementById("new-number").value;
+  row.innerHTML = `<tr class="product"><td class="name"><span>${newName}</span></td>
+  <td class="pu">$<span>${newPu}</span></td><td class="qty"><label><input type="number" value="0" min="0">
+  </label></td><td class="subtot">$<span>0</span></td><td class="rm"><button class="btn btn-delete">Delete</button>
+  </td></tr>`
+}
```

Comment:
A bit of hardcoding here, but it's fine since the challenge is difficult. ;)

As an example, there is also this other approach
`function createNewProduct() {
  let createDiv = document.querySelector("#cart > tfoot > tr"); // This is the "create new product div"
  let name = createDiv.querySelector("td:nth-child(1) > input").value; // This is the name input for the new product
  let priceUnit = createDiv.querySelector("td:nth-child(2) > input").value; // This is the price per unit input for the new product
  let newTrElement = document
    .querySelector("#cart > tbody > tr:nth-child(1)")
    .cloneNode(true); // this create a copy of the <tr> product element
  newTrElement.querySelector("td.name > span").innerHTML = name; // sets the name of the new product
  newTrElement.querySelector("td.pu > span").innerHTML = priceUnit; // sets the price per unit of the new product
  document.querySelector("#cart > tbody").appendChild(newTrElement); // adds the element to the carts tbody
}`